### ISSUE TITLE
Separate the "markdown" from the "slideshow"

### ISSUE THREAD

**Issue by alex-rozenshteyn-leapyear** [gnab/remark#644]
I would like to be able to generate a self-contained HTML doc _with no markdown_ and present that. All of the tools in the "static generators" section just embed a markdown doc in the template HTML.

I like the slideshow generated by `remark` more than any other HTML slideshow tool I've found so far, especially with the easy theming and the presenter view, but it would be nice to be able to use it as an HTML presentation tool and not just a markdown presentation tool.

**Comment by abelards** [comment]
Hi Alex, I'm not sure what your feelings are bout this.
- remark is by definition a markdown tool, it's in the name
- you would need at least some of remark's markdown to separate slides
- you can add as much HTML as you want, markdown does nothing with it and passes it on directly

So... if you're keeping that minimal markdown and you put as much HTML as you want, what's the problem?
Is "self-contained HTML doc" some kind of huge page with slides' thumbnails and presenter notes?

Let us know.
Cheers,

**Comment by alex-rozenshteyn-leapyear** [comment]
> remark is by definition a markdown tool, it's in the name

Yes, but it's not _only_ a markdown tool; it's _also_ a presentation runtime (unless I'm misunderstanding its architecture).

> Is "self-contained HTML doc" some kind of huge page with slides' thumbnails and presenter notes?

Yes. I'm basically imagining the DOM that remark generates, or some simplification of it.

I'm basically imagining a way to use remark as a lightweight opinionated version of [reveal.js](https://revealjs.com/) while delegating the markdown translation to something else.

Thanks for the reply.

**Comment by seasonedbits** [comment]
@alex-rozenshteyn-leapyear I think you're suggesting to seperate the markup layer from the presentation layer, right?
In that way we can "supposedly" turn any markup language into preseatation.

Your suggestion requires a "meta" markup which can store slide meta (layout, attributes, seperator, etc) outside of the content markup.
As @abelards said, that is not the aim for remark. remark will always support markdown and have used (or hiajcked one may even say) markdown's syntax as slide seperator. It have its own syntax based on markdown.
Since HTML is valid markdown, you can use markdown or plain HTML in your slide.

**Comment by alex-rozenshteyn-leapyear** [comment]
> @alex-rozenshteyn-leapyear I think you're suggesting to seperate the markup layer from the presentation layer, right?
> In that way we can "supposedly" turn any markup language into preseatation.

I think this is broadly accurate. Digging in one step deeper, I want to extract the presentation layer, so that I can use it without the markup layer (I'd like to do my own markdown processing).

**Comment by seasonedbits** [comment]
I think you'll better luck looking for "presentation layout" for static site generators.
Try Next.js, Eleventy, Hexo and the likes.
They'll be more flexible in processing content files.
